File: app/src/main/java/com/waygo/data/model/fuel/Premium.java
package com.waygo.data.model.fuel;


import android.support.annotation.NonNull;

import com.waygo.utilskt.Result;


public final class Premium extends Fuel {
    protected Premium(float value) {
        super(value);
    }

    @NonNull
    public static Result<Fuel> create(final float value) {
        return createValue(value).map(v -> (Fuel) new Premium(v));
    }
}

File: app/src/main/java/com/waygo/data/model/fuel/Premium.kt
package com.waygo.data.model.fuel

import com.waygo.utilskt.Result
import kotlin.platform.platformStatic

public class Premium protected constructor(value: Float) : Fuel(value) {
    companion object {
        public platformStatic fun create(value: Float): Result<Fuel> =
                Fuel.createValue(value)
                        .map({ Premium })
                        .ofType(javaClass<Fuel>())
    }
}
